src/rgw/rgw_rados.cc

```diff
@@ -8741,6 +8741,8 @@ int RGWRados::get_obj_state_impl(RGWObjectCtx *rctx, const RGWBucketInfo& bucket
       s->manifest.set_head(bucket_info.placement_rule, obj, s->size); /* patch manifest to reflect the head we just read, some manifests might be
                                              broken due to old bugs */
       s->size = s->manifest.get_obj_size();
+      if (s->accounted_size < s->size)
```

Comment:
this is making sense to me now. `accounted_size` defaults to `size` above, before looking at the `RGWCompressionInfo`. if we update `size` here, that update should also be applied to `accounted_size` as well

things are tricky if -both- `accounted_size` is set from `RGWCompressionInfo` -and- `size` is set based on the manifest. while it's possible for compression to increase the object size, i think it's safe to report to smaller of the two in this unlikely case